stores/sql/main.go

```diff
@@ -2,13 +2,95 @@ package sql
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"strings"
 
+	dsql "database/sql"
+
 	"go.sia.tech/core/types"
+	"go.sia.tech/renterd/api"
 	"go.sia.tech/renterd/internal/sql"
 )
 
+func Contracts(ctx context.Context, tx sql.Tx, opts api.ContractsOpts) ([]api.ContractMetadata, error) {
+	var rows *sql.LoggedRows
+	var err error
+	if opts.ContractSet != "" {
+		// if we filter by contract set, we fetch the set first to check if it
+		// exists and then fetch the contracts. Knowing that the contracts are
+		// part of at least one set allows us to use INNER JOINs.
+		var contractSetID int64
+		err = tx.QueryRow(ctx, "SELECT id FROM contract_sets WHERE contract_sets.name = ?", opts.ContractSet).
+			Scan(&contractSetID)
+		if errors.Is(err, dsql.ErrNoRows) {
+			return nil, api.ErrContractSetNotFound
+		}
+		rows, err = tx.Query(ctx, `
+			SELECT c.fcid, c.renewed_from, c.contract_price, c.state, c.total_cost, c.proof_height,
+			c.revision_height, c.revision_number, c.size, c.start_height, c.window_start, c.window_end,
+			c.upload_spending, c.download_spending, c.fund_account_spending, c.delete_spending, c.list_spending,
+			cs.name, h.net_address, h.public_key, h.settings->>'$.siamuxport' AS siamux_port
```

Comment:
would use `JSON_EXTRACT`, isn't `->>` a PostgreSQL thing?

Reply:
`JSON_EXTRACT` is not doing the same on SQLite and MySQL. On MySQL it returns the `""` too. `->>` is the same on both though.